### PR TITLE
Add circuit breaker for relayer/builder

### PR DESCRIPTION
**Pull request by terencechain**
This PR implements a circuit breaker for the builder/relayer infra to determine whether or not we should propose blocks through the builder API. This code doesn't live on the normal block proposal path, it lives in the builder block proposal path, and the risk factor is low. With that said, please careful review 

The following flags are added:
`--builder-fallback-skips` (default value 4)
the number of slots before the proposal slot that has been skipped
`--builder-fallback-skips-per-epoch` (default value 6)
the number of slots in the last 32 slots that have been skipped

Would appreciate feedback on the defaults!

### CODE REVIEW

File: beacon-chain/node/config.go [hunk 1]
```diff
@@ -65,6 +65,24 @@ func configureSafeSlotsToImportOptimistically(cliCtx *cli.Context) error {
 	return nil
 }
 
+func configureMevBoostCircuitBreakerValues(cliCtx *cli.Context) error {
```
Comment (james-prysm) [inline beacon-chain/node/config.go:68]: maybe we should name it form `configureMevBoostCircuitBreakerValues` to`configureBuilderCircuitBreaker`

File: beacon-chain/node/config.go [hunk 2]
```diff
@@ -65,6 +65,24 @@ func configureSafeSlotsToImportOptimistically(cliCtx *cli.Context) error {
...
+	if cliCtx.IsSet(flags.BuilderFallbackSkips.Name) {
+		c := params.BeaconConfig().Copy()
+		c.BuilderFallbackSkipsSlot = types.Slot(cliCtx.Int(flags.BuilderFallbackSkips.Name))
```
Comment (james-prysm) [inline beacon-chain/node/config.go:71]: maybe rename to `MaxBuilderConsecutiveMissedSlots` telling the system that there should be a max of x slots missed before falling back to something

File: beacon-chain/node/config.go [hunk 3]
```diff
@@ -65,6 +65,24 @@ func configureSafeSlotsToImportOptimistically(cliCtx *cli.Context) error {
...
+		if err := params.SetActive(c); err != nil {
+			return err
+		}
+	}
+	if cliCtx.IsSet(flags.BuilderFallbackSkipsPerEpoch.Name) {
+		c := params.BeaconConfig().Copy()
+		c.BuilderFallbackSkipsSlotsLastEpoch = types.Slot(cliCtx.Int(flags.BuilderFallbackSkipsPerEpoch.Name))
```
Comment (james-prysm) [inline beacon-chain/node/config.go:78]: and corresponding `MaxBuilderMissedSlotsInEpoch`

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go [hunk 4]
```diff
@@ -313,6 +361,15 @@ func (vs *Server) getAndBuildBlindBlock(ctx context.Context, b *ethpb.BeaconBloc
 	if !ready {
 		return false, nil, nil
 	}
+
+	circuitBreak, err := vs.circuitBreakBuilder(b.Slot)
+	if err != nil {
+		return false, nil, errors.Wrap(err, "could not determine if builder circuit breaker condition")
```
Comment (james-prysm) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go:367]: maybe remove the if statement

File: cmd/beacon-chain/flags/base.go [hunk 5]
```diff
@@ -16,6 +16,16 @@ var (
 		Usage: "A MEV builder relay string http endpoint, this wil be used to interact MEV builder network using API defined in: https://ethereum.github.io/builder-specs/#/Builder",
 		Value: "",
 	}
+	BuilderFallbackSkips = &cli.IntFlag{
+		Name:  "builder-fallback-skips",
+		Usage: "Number of consecutive skip slot to fallback from using relay/builder to local execution engine for block construction",
```
Comment (james-prysm) [inline cmd/beacon-chain/flags/base.go:21]: related to flag name changes

**Review by james-prysm — APPROVED** [review]
Approving but might want a second pair of eyes maybe?

**Review by rauljordan — APPROVED** [review]

**Comment by rauljordan** [comment]
Really nice PR, thanks for putting it together in a concise manner

**Pull request merged**